platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/ConfigurationAdminTest.java

```diff
@@ -729,24 +769,83 @@ public void testSetBundleLocationNullLocation() throws Exception {
-        Entry<String, Object> testEntry1 = mock(Entry.class);
-        testEntrySet.add(testEntry1);
-
-        when(testEntry1.getValue()).thenReturn(TEST_VALUE);
-
-        when(testConfigAdmin.getConfiguration(TEST_PID, null)).thenReturn(testConfig);
+        ConfigurationAdmin configAdmin = getConfigAdmin();
+        // test every typed cardinality<->cardinality mapping
+        for (int i = 0; i < CARDINALITIES.length; i++) {
+            int cardinality = CARDINALITIES[i];
+            Map<String, Object> testConfigTable = new Hashtable<>();
+            for (TYPE type : TYPE.values()) {
+                for (int keyCardinality : CARDINALITIES) {
+                    testConfigTable.put(getKey(keyCardinality, type), getValue(cardinality, type));
+                }
+            }
+            configAdmin.update(TEST_PID, testConfigTable);
+            verify(testConfig, times(i + 1)).update(any(Dictionary.class));
+        }
+        Hashtable<String, Object> values = new Hashtable<>();
+        String key = getKey(CARDINALITY_ARRAY, TYPE.STRING);
+        values.put(key, "[\"foo\",\"bar\",\"baz\"]");
+        configAdmin.update(TEST_PID, values);
+        ArgumentCaptor<Dictionary> captor = ArgumentCaptor.forClass(Dictionary.class);
+        verify(testConfig, times(4)).update(captor.capture());
+        assertThat(((String[]) captor.getValue().get(key)).length, equalTo(3));
+        key = getKey(CARDINALITY_PRIMITIVE, TYPE.BOOLEAN);
+        values.put(key, "true");
+        configAdmin.update(TEST_PID, values);
+        verify(testConfig, times(5)).update(captor.capture());
+        assertThat((Boolean) captor.getValue().get(key), equalTo(true));
+    }
 
-        configAdmin.update(TEST_PID, testConfigTable);
-        verify(testConfig).update(any(Dictionary.class));
+    private Object getValue(int cardinality, TYPE type) {
+        Object value = null;
+        switch (type) {
+        case PASSWORD:
+        case STRING:
+            value = TEST_VALUE;
+            break;
+        case BIGDECIMAL:
+            value = BigDecimal.valueOf(TEST_INT);
+            break;
+        case BIGINTEGER:
+            value = BigInteger.valueOf(TEST_INT);
+            break;
+        case BOOLEAN:
+            value = true;
+            break;
+        case BYTE:
+            value = (byte) TEST_INT;
+            break;
+        case CHARACTER:
+            value = 'c';
+            break;
+        case DOUBLE:
+            value = (double) TEST_INT;
+            break;
+        case FLOAT:
+            value = (float) TEST_INT;
+            break;
+        case INTEGER:
+            value = TEST_INT;
+            break;
+        case LONG:
+            value = (long) TEST_INT;
+            break;
+        case SHORT:
+            value = (short) TEST_INT;
+            break;
+        }
+        switch (cardinality) {
+        case CARDINALITY_VECTOR:
+            Vector<Object> vector = new Vector<>();
```

Comment:
Strictly speaking, the only allowable values wrt cardinality are array, primitive, and vector. However, since vector is a collection and the implementation only checks for arrays or collections, it's more of a point of philosophical purity.

In other words, there's no hard reason other than that vector is what a negative cardinality represents, so I was just being consistent.